lib/ain-contracts/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+use anyhow::anyhow;
+use lazy_static::lazy_static;
+use primitive_types::{H160, H256, U256};
+use std::collections::HashMap;
+use std::error::Error;
+use std::str::FromStr;
+
+pub fn u256_to_h256(input: U256) -> H256 {
+    let mut bytes = [0_u8; 32];
+    input.to_big_endian(&mut bytes);
+
+    H256::from(bytes)
+}
+
+pub fn get_abi_encoded_string(input: &str) -> H256 {
+    let length = input.len();
+
+    let mut storage_value = H256::default();
+    storage_value.0[31] = (length * 2) as u8;
+    storage_value.0[..length].copy_from_slice(input.as_bytes());
+
+    storage_value
+}
+
+pub fn get_bytecode(input: &str) -> Result<Vec<u8>, Box<dyn Error>> {
```

Comment:
Might want to add in rust debug logs and errors to propagate them downstream.

Reply:
The errors are already being propagated here it seems. Not sure what you mean about the logs :/

Reply:
The code already does this.